google/cloud/storage/examples/storage_bucket_default_kms_key_samples.cc

```diff
@@ -0,0 +1,161 @@
+              << updated->encryption().default_kms_key_name << "."
+              << "\nFull metadata: " << *updated << "\n";
+  }
+  //! [add bucket kms key] [END storage_set_bucket_default_kms_key]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void GetBucketDefaultKmsKey(google::cloud::storage::Client client,
+                            std::vector<std::string> const& argv) {
+  //! [get bucket default kms key] [START storage_bucket_get_default_kms_key]
+  namespace gcs = google::cloud::storage;
+  using ::google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name) {
+    StatusOr<gcs::BucketMetadata> metadata =
+        client.GetBucketMetadata(bucket_name);
+    if (!metadata) throw std::runtime_error(metadata.status().message());
+
+    if (!metadata->has_encryption()) {
+      std::cout << "The bucket " << metadata->name()
+                << " does not have a default KMS key set.\n";
+      return;
+    }
+
+    std::cout << "The default KMS key for bucket " << metadata->name()
+              << " is: " << metadata->encryption().default_kms_key_name << "\n";
+  }
+  //! [get bucket default kms key] [END storage_bucket_get_default_kms_key]
+  (std::move(client), argv.at(0));
+}
+
+void RemoveBucketDefaultKmsKey(google::cloud::storage::Client client,
+                               std::vector<std::string> const& argv) {
+  //! [remove bucket default kms key]
+  // [START storage_bucket_delete_default_kms_key]
+  namespace gcs = google::cloud::storage;
+  using ::google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name) {
+    StatusOr<gcs::BucketMetadata> updated = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().ResetEncryption());
+    if (!updated) throw std::runtime_error(updated.status().message());
+
+    std::cout << "Successfully removed default KMS key on bucket "
+              << updated->name() << "\n";
+  }
+  // [END storage_bucket_delete_default_kms_key]
+  //! [remove bucket default kms key]
+  (std::move(client), argv.at(0));
+}
+
+void RunAll(std::vector<std::string> const& argv) {
+  if (!argv.empty()) throw Usage{"auto"};
+
+  namespace examples = ::google::cloud::storage::examples;
+  namespace gcs = ::google::cloud::storage;
+
+  examples::CheckEnvironmentVariablesAreSet({
+      "GOOGLE_CLOUD_PROJECT",
+      "GOOGLE_CLOUD_CPP_STORAGE_TEST_CMEK_KEY",
+  });
+  auto const project_id =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value();
+  auto const cmek_key =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_STORAGE_TEST_CMEK_KEY")
+          .value();
+  auto generator = google::cloud::internal::DefaultPRNG(std::random_device{}());
+  auto const bucket_name =
+      examples::MakeRandomBucketName(generator, "cloud-cpp-test-examples-");
+  auto client = gcs::Client::CreateDefaultClient().value();
+
+  std::cout << "\nCreating bucket to run the example (" << bucket_name << ")"
+            << std::endl;
```

Comment:
These `std::endl`s look a little out of place. Consider using "\n" at the end of the preceding string, which I think is our normal style. And if flushing is needed, consider using `std::flush` to make that clear.

Reply:
We have been using `std::endl` in all the examples. Yes, flushing is needed because when an example crashes you want to know which one was it.

If it is all the same, I am going to stick to `std::endl`, I think it is clear enough.

Reply:
If that's what we've been doing, then OK.